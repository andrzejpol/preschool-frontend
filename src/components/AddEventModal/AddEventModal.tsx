import React from "react";
import {useDispatch} from "react-redux";
import * as yup from "yup";
import {useForm} from "react-hook-form";
import {yupResolver} from "@hookform/resolvers/yup";
import {v4 as uuidv4} from "uuid";
import {closeModal} from "../../slices/modalsSlice.ts";

const AddEventModal = ({onClose}: { onClose: () => {} }) => {
  const dispatch = useDispatch();

  const schema = yup.object({
    id: yup.string().required(),
    title: yup.string().required("Title is required"),
    group: yup.string().required("Group is required"),
    date: yup.date().required("Date is required"),
    startTime: yup.string().required("Start time is required"),
    endTime: yup.string().required("End time is required"),
  });

  const {
    handleSubmit,
    setValue,
    register,
    reset,
    formState: {errors},
  } = useForm({
    resolver: yupResolver(schema),
    defaultValues: {
      id: uuidv4(),
      title: "",
      group: "",
      date: new Date(),
      startTime: "",
      endTime: "",
    },
  });

  const onSubmit = (data: any) => {
    // dispatch(addEvent(data));
    // reset();
    // dispatch(closeModal({modalName: 'addEventModal'}));
    alert("Modal ")
  };

  return (
    <div onClick={onClose} className="fixed inset-0 flex items-center justify-center bg-gray-900 bg-opacity-50">
      <div onClick={(e) => e.stopPropagation()}
           className="bg-white w-11/12 max-w-lg p-6 rounded-lg shadow-lg relative">
        <button
          onClick={onClose}
          className="absolute top-4 right-4 text-gray-500 hover:text-gray-700 focus:outline-none"
        >
          âœ–
        </button>
        <h2 className="text-2xl font-bold mb-4 text-center">Add event</h2>
        <form onSubmit={handleSubmit(onSubmit)}>
          <div className="mb-4">
            <label htmlFor="title" className="block text-gray-700 font-medium mb-2">
              Title:
            </label>
            <input
              type="text"
              id="title"
              {...register("title")}
              className="w-full border border-gray-300 rounded p-2 bg-white"
            />
            {errors.title && (
              <p className="text-red-500 text-sm mt-1">{errors.title.message}</p>
            )}
          </div>
          <div className="mb-4">
            <label htmlFor="group" className="block text-gray-700 font-medium mb-2">
              Group:
            </label>
            <select
              id="group"
              {...register("group")}
              className="w-full border border-gray-300 rounded p-2 bg-white"
            >
              <option value="" disabled>
                Select a group
              </option>
              <option value="Group 1">Group 1</option>
              <option value="Group 2">Group 2</option>
              <option value="Group 3">Group 3</option>
            </select>
            {errors.group && (
              <p className="text-red-500 text-sm mt-1">{errors.group.message}</p>
            )}
          </div>
          <div className="mb-4">
            <label htmlFor="date" className="block text-gray-700 font-medium mb-2">
              Date:
            </label>
            <input
              type="date"
              id="date"
              {...register("date")}
              className="w-full border border-gray-300 rounded p-2 bg-white"
            />
            {errors.date && (
              <p className="text-red-500 text-sm mt-1">{errors.date.message}</p>
            )}
          </div>
          <div className="mb-4 flex space-x-4">
            <div className="mb-4">
              <label htmlFor="startTime" className="block text-gray-700 font-medium mb-2">
                Start time:
              </label>
              <input
                type="time"
                id="startTime"
                {...register("startTime")}
                className="w-full border border-gray-300 rounded p-2 bg-white"
              />
              {errors.startTime && (
                <p className="text-red-500 text-sm mt-1">{errors.startTime.message}</p>
              )}
            </div>
            <div className="mb-4">
              <label htmlFor="endTime" className="block text-gray-700 font-medium mb-2">
                End Time:
              </label>
              <input
                type="time"
                id="endTime"
                {...register("endTime")}
                className="w-full border border-gray-300 rounded p-2 bg-white"
              />
              {errors.endTime && (
                <p className="text-red-500 text-sm mt-1">{errors.endTime.message}</p>
              )}
            </div>
          </div>
          <button
            type="submit"
            className="bg-blue-500 text-white px-4 py-2 rounded hover:bg-blue-600"
          >
            Add
          </button>
        </form>
      </div>
    </div>
  )
};

export default AddEventModal;