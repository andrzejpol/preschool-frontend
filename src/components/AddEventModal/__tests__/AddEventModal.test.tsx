import React from "react";
import {render} from "@testing-library/react";
import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import AddEventModal from "../AddEventModal";

const mockStore = configureStore([]);

describe("AddEventModal Component", () => {
  let store;

  beforeEach(() => {
    store = mockStore({});
  });

  it("matches the snapshot", () => {
    const {asFragment} = render(
      <Provider store={store}>
        <AddEventModal onClose={vi.fn()}/>
      </Provider>
    );

    // Snapshot assertion
    expect(asFragment()).toMatchSnapshot();
  });
});
