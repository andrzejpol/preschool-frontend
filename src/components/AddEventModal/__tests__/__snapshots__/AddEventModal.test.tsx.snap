// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddEventModal Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="fixed inset-0 flex items-center justify-center bg-gray-900 bg-opacity-50"
  >
    <div
      class="bg-white w-11/12 max-w-lg p-6 rounded-lg shadow-lg relative"
    >
      <button
        class="absolute top-4 right-4 text-gray-500 hover:text-gray-700 focus:outline-none"
      >
        âœ–
      </button>
      <h2
        class="text-2xl font-bold mb-4 text-center"
      >
        Add event
      </h2>
      <form>
        <div
          class="mb-4"
        >
          <label
            class="block text-gray-700 font-medium mb-2"
            for="title"
          >
            Title:
          </label>
          <input
            class="w-full border border-gray-300 rounded p-2 bg-white"
            id="title"
            name="title"
            type="text"
          />
        </div>
        <div
          class="mb-4"
        >
          <label
            class="block text-gray-700 font-medium mb-2"
            for="group"
          >
            Group:
          </label>
          <select
            class="w-full border border-gray-300 rounded p-2 bg-white"
            id="group"
            name="group"
          >
            <option
              disabled=""
              value=""
            >
              Select a group
            </option>
            <option
              value="Group 1"
            >
              Group 1
            </option>
            <option
              value="Group 2"
            >
              Group 2
            </option>
            <option
              value="Group 3"
            >
              Group 3
            </option>
          </select>
        </div>
        <div
          class="mb-4"
        >
          <label
            class="block text-gray-700 font-medium mb-2"
            for="date"
          >
            Date:
          </label>
          <input
            class="w-full border border-gray-300 rounded p-2 bg-white"
            id="date"
            name="date"
            type="date"
          />
        </div>
        <div
          class="mb-4 flex space-x-4"
        >
          <div
            class="mb-4"
          >
            <label
              class="block text-gray-700 font-medium mb-2"
              for="startTime"
            >
              Start time:
            </label>
            <input
              class="w-full border border-gray-300 rounded p-2 bg-white"
              id="startTime"
              name="startTime"
              type="time"
            />
          </div>
          <div
            class="mb-4"
          >
            <label
              class="block text-gray-700 font-medium mb-2"
              for="endTime"
            >
              End Time:
            </label>
            <input
              class="w-full border border-gray-300 rounded p-2 bg-white"
              id="endTime"
              name="endTime"
              type="time"
            />
          </div>
        </div>
        <button
          class="bg-blue-500 text-white px-4 py-2 rounded hover:bg-blue-600"
          type="submit"
        >
          Add
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
